core/src/main/java/uk/gov/gchq/koryphe/iterable/ChainedIterator.java

```diff
@@ -63,12 +65,16 @@ private Iterator<? extends T> getIterator() {
         while (!currentIterator.hasNext()) {
             CloseableUtil.close(currentIterator);
             if (iterablesIterator.hasNext()) {
-                currentIterator = iterablesIterator.next().iterator();
+                Object next = iterablesIterator.next();
+                if (nonNull(next) && next instanceof Iterable) {
+                    currentIterator = (Iterator<? extends T>) ((Iterable<?>) next).iterator();
```

Comment:
Could this also be `(Iterable<? extends T>)` or am I misunderstanding?

Reply:
```suggestion
                    currentIterator = ((Iterable<? extends T>) next).iterator();
```